### ISSUE TITLE
Rename removeF to removeBy

### ISSUE THREAD

**Issue by mlms13** [reazen/relude#28]
...and all of the other `...F` suffixed functions. This seems to be consistent with PureScript's `maximumBy` where you pass in an `a -> a -> Ordering` instead of a member of Ord.

Initially I was thinking that the function-based (as opposed to module-based) versions should be the "default" (e.g. `eq` would be the function-based version instead of using an Eq module), but prioritizing modules feels "more correct", and I don't mind the `...By` suffix quite as much.

**Comment by arecvlohe** [comment]
This would just be for ordering? I see there is a `showF` and `SHOW_F`. Just want to make sure before I take a stab at updating this.

**Comment by arecvlohe** [comment]
Oh I see it's being worked on 😄

**Comment by andywhite37** [comment]
@arecvlohe - yeah I started working on this one before I saw your comment.  Just to explain, for functions like `eq` in some of our types like List/Array/Result/etc., we have a version of the function that takes an `Eq` module for the contained type, and a function that takes a simple comparison function (not wrapped in a module).  At some point, I think we decided we wanted to prefer the module-based functions, pretending that modular implicits exist, so we used the preferred name `eq` for the function that takes the `module`.  The function that takes the comparison function was given the name `eqF` (for equals with a function), but we never liked that name.  This effort was to rename `eqF` to `eqBy`, but you make a good point that other `*F` functions should also be renamed to `*By`.  I can take care of this one for now.
